zerver/lib/test_classes.py

```diff
@@ -2420,37 +2422,38 @@ def check_webhook(
             headers = get_fixture_http_headers(self.WEBHOOK_DIR_NAME, fixture_name)
             headers = standardize_headers(headers)
             extra.update(headers)
-        try:
-            msg = self.send_webhook_payload(
-                self.test_user,
-                self.url,
-                payload,
-                **extra,
-            )
-        except EmptyResponseError:
-            if expect_noop:
-                return
-            else:
-                raise AssertionError(
-                    "No message was sent. Pass expect_noop=True if this is intentional."
+        for url in [*self.legacy_urls, self.url]:
```

Comment:
`check_webhook()` now tests all urls supported by a webhook.
Just wrapped inside a for loop.